### PR TITLE
fix refetching logs

### PR DESCRIPTION
**Pull request by et**
## Summary

This PR fixes two frontend issues (see inline comments below).

## How did you test this change?

See inline comments below.

## Are there any deployment considerations?

N/A

### CODE REVIEW

File: frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ export const LogsTable = ({ data, loading, query }: Props) => {
 	useEffect(() => {
 		// Collapse all rows when search changes
 		table.toggleAllRowsExpanded(false)
-	}, [logEdges])
+	}, [logEdges, table])
```
Comment (et) [inline frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx:111]: See https://github.com/highlight/highlight/pull/4393#discussion_r1119002239

Prior to this change, we had an infinite loop:

Screenshot 2023-02-27 at 4 43 45 PM

After this change, the error goes away.

File: frontend/src/pages/LogsPage/LogsPage.tsx [hunk 2]
```diff
@@ -75,7 +75,7 @@ const LogsPage = () => {
 			},
 		},
 		skip: !project_id,
-		nextFetchPolicy: 'cache-first', // See: https://stackoverflow.com/a/66489126
+		fetchPolicy: 'cache-and-network',
```
Comment (et) [inline frontend/src/pages/LogsPage/LogsPage.tsx:78]: Prior to this change, when we updated the query input or set the time range, the request to fetch new logs wouldn't be run. 

There's not too much written about `cache-and-network` but it is what is recommended to use from the [docs](https://www.apollographql.com/docs/react/v2/caching/cache-interaction/#incremental-loading-fetchmore).

Comment (et) [reply frontend/src/pages/LogsPage/LogsPage.tsx:78]: I confirmed that forward scrolling works _and_ it's not making a double request (what the previous stackoverflow comment was addressing).

**Review by Vadman97 — APPROVED** [review]
nice find

**Pull request merged**